### PR TITLE
Remove the suppression of rawtypes warnings.

### PR DESCRIPTION
**Pull request by wiarlawd**
There were five cases:

* Code that expected generic Objects (use a wildcard type parameter)
* Unnecessary type parameters on a class (remove parameterization)
* Unnecessary type parameters on a method (use inferred type)
* Incomplete type parameters on a method (add full type parameters)
* Overrides of API methods that do not use generics (suppress warning)

### CODE REVIEW

File: src/com/google/enterprise/adaptor/CronScheduler.java [hunk 1]
```diff
@@ -72,7 +72,7 @@ public void reschedule(Future<?> future, String pattern) {
     if (!(future instanceof CronFuture)) {
       throw new IllegalArgumentException("Provided future is not a CronFuture");
     }
-    CronFuture cronFuture = (CronFuture) future;
+    CronFuture<?> cronFuture = (CronFuture) future;
```
Comment (PapaJoltOmega) [inline src/com/google/enterprise/adaptor/CronScheduler.java:75]: The updated line adds "<?>".  

This addition seems to make less readable to me.  

What do you think?

**Review by PapaJoltOmega — COMMENTED** [review]
Thank you

File: src/com/google/enterprise/adaptor/CronScheduler.java (hunk 1 above)
Comment (wiarlawd) [reply src/com/google/enterprise/adaptor/CronScheduler.java:75]: I think it's the language we have. I really dislike suppressing warnings. It's harmless in cases like this, but not everything that could get suppressed is harmless, and with the warnings suppressed we write code we didn't mean to (e.g., the List<Map.Entry> stuff in DocIdSenderTest).

Comment (PapaJoltOmega) [reply src/com/google/enterprise/adaptor/CronScheduler.java:75]: What's wrong with List<Map.Entry> in DocIdSenderTest?

**Review by PapaJoltOmega — COMMENTED** [review]
What's wrong with List<Map.Entry> in DocIdSenderTest?

Thank you

**Review by PapaJoltOmega — APPROVED** [review]

**Pull request merged**